File: reverseBits.ts
{
    const reverseBits = function(n) {
        let res = 0, 
            pow = 31;

        while (n > 0) {
            const rightMost = n & 1;
            res = res + (rightMost << pow);
            pow--;
            n = n >>> 1;
        }

        return res >>> 0;
    };
}

File: reverse-bits.rs
pub fn reverse_bits(n: i32) -> i32 {
    let mut ans: i32 = 0;
    for i in 0..32 {
        if n & (1 << i) != 0 {
            ans = ans | 1 << (31 - i);
        }
    }
    ans
}
